injector/grpc_disruption_test.go

```diff
@@ -0,0 +1,134 @@
+// Unless explicitly stated otherwise all files in this repository are licensed
+// under the Apache License Version 2.0.
+// This product includes software developed at Datadog (https://www.datadoghq.com/).
+// Copyright 2021 Datadog, Inc.
+
+/*
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+    http://www.apache.org/licenses/LICENSE-2.0
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package injector_test
+
+import (
+	"testing"
+
+	"github.com/DataDog/chaos-controller/api/v1beta1"
+	"github.com/DataDog/chaos-controller/grpc"
+	. "github.com/DataDog/chaos-controller/injector"
+
+	pb "github.com/DataDog/chaos-controller/grpc/disruption_listener"
+	. "github.com/onsi/ginkgo"
+	. "github.com/onsi/gomega"
+	"github.com/stretchr/testify/mock"
+	"google.golang.org/protobuf/types/known/emptypb"
+)
+
+func TestSendAndCleaDisruption(t *testing.T) {
+	RegisterFailHandler(Fail)
+	RunSpecs(t, "gRPC Disruption Suite - Send Disruption")
+
+	// define parameters of NewGRPCDisruptionInjector
+	spec := v1beta1.GRPCDisruptionSpec{
+		Port: 2000,
+		Endpoints: []v1beta1.EndpointAlteration{
+			{
+				TargetEndpoint:   "/chaos_dogfood.ChaosDogfood/order",
+				ErrorToReturn:    "",
+				OverrideToReturn: "{}",
+				QueryPercent:     50,
+			},
+			{
+				TargetEndpoint:   "/chaos_dogfood.ChaosDogfood/getCatalog",
+				ErrorToReturn:    "NOT_FOUND",
+				OverrideToReturn: "",
+				QueryPercent:     25,
+			},
+			{
+				TargetEndpoint:   "/chaos_dogfood.ChaosDogfood/getCatalog",
+				ErrorToReturn:    "ALREADY_EXISTS",
+				OverrideToReturn: "",
+				QueryPercent:     50,
+			},
+			{
+				TargetEndpoint:   "/chaos_dogfood.ChaosDogfood/getCatalog",
+				ErrorToReturn:    "",
+				OverrideToReturn: "{}",
+				QueryPercent:     0,
+			},
+		},
+	}
+
+	disruptionListenerClient := &DisruptionListenerClientMock{}
+
+	// define expectations
```

Comment:
What does this file test?

Reply:
I want to test that a fairly common script calls `SendDisruption` with the expected parameters using gRPC without actually making the call. This is not encapsulated by our current `e2e` test which only checks that the pod lifecycles are as expected. As our `e2e` gets more sophisticated, we likely won't need this test anymore. I'm also okay to remove this file if we feel ready to do so!